tests/unit/test_artifactory_path.py

```diff
@@ -967,5 +971,190 @@ def test_from_aql_file(self):
         assert artifact.root == "/reponame/"
 
 
+class TestArtifactoryPathGetAll(unittest.TestCase):
+    # TODO: test repositories and permissions
+    def setUp(self):
+        self.arti = ArtifactoryPath("http://b.com/artifactory")
+
+    def test_get_all_users(self):
+        url = "/api/security/users"
+        request_url = self.arti.drive + url
+        key = "name"
+        cls = User
+        users = [
+            {
+                "name": "user_1",
+                "uri": "http://b.com/artifactory/api/security/users/user_1",
+                "realm": "internal",
+            },
+            {
+                "name": "user_2",
+                "uri": "http://b.com/artifactory/api/security/users/user_2",
+                "realm": "internal",
+            },
+        ]
+        user_1 = {"name": "user_1", "email": "user1@example.com"}
+        user_2 = {"name": "user_2", "email": "user2@example.com"}
+
+        # lazy = True
+        with responses.RequestsMock() as rsps:
+            rsps.add(responses.GET, request_url, json=users, status=200)
+            results = self.arti._get_all(lazy=True, url=url, key=key, cls=cls)
+
+            map(self.assertIsInstance, zip(results, repeat(cls)))
+            self.assertEqual(results[0].name, "user_1")
+            self.assertIsNone(results[0].email)
```

Comment:
The asserts about the instance should be inside the for loop too

Reply:
Fixed